File: Backend/TranscomaAPI/Comun/Entidades/Salida.cs
using System;

namespace TranscomaAPI.Comun.Entidades
{
    public class Salida : Entidad
    {
        private DateTime _fechaSalida;
        private Producto _producto;
        private Almacen _almacen;

        /// <summary>
        /// 
        /// </summary>
        /// <param name="id"></param>
        public Salida(int id)
        {
            Id = id;
        }

        /// <summary>
        /// Registro de la fecha de Salida
        /// </summary>
        /// <param name="fechaSalida"></param>
        public Salida(DateTime fechaSalida)
        {
            _fechaSalida = fechaSalida;
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="id"></param>
        /// <param name="fechaSalida"></param>
        public Salida(int id, DateTime fechaSalida)
        {
            Id = id;
            _fechaSalida = fechaSalida;
        }

        /// <summary>
        /// Getters y Setters
        /// </summary>
        public DateTime FechaSalida { get => _fechaSalida; set => _fechaSalida = value; }
        public Producto Productos { get => _producto; set => _producto = value; }
        public Almacen Almacenes { get => _almacen; set => _almacen = value; }
    }
}

File: frontend/src/app/models/Salida.ts
import { EntidadBase } from "./EntidadBase";
import { Cliente } from "./Cliente";
import { Producto } from "./Producto";
import { Almacen } from "./Almacen";

export class Salida extends EntidadBase {

    private _fechaSalida: string;
    private _producto: Producto;
    private _almacen: Almacen;
    private _cliente: Cliente;

    constructor(data?: any) {
        super();
        if (data !== undefined) {
            this.$id = data.id;
            this.$fechaSalida = data.fechaSalida;
            // this.$producto = data.producto;
            // this.$almacen = data.almacen;
            // this.$cliente = data.cliente;

            if (data.producto != undefined) {
                this.$producto = new Producto(data.producto);
            }
            else {
                this.$producto = null;
            }

            if (data.almacen != undefined) {
                this.$almacen = new Almacen(data.almacen);
            }
            else {
                this.$almacen = null;
            }

            if (data.cliente != undefined) {
                this.$cliente = new Cliente(data.cliente);
            }
            else {
                this.$cliente = null;
            }
        }
    }

    /** Getters y setters */
    public get $fechaSalida(): string {
        return this._fechaSalida;
    }

    public set $fechaSalida(value: string) {
        this._fechaSalida = value;
    }

    public get $producto(): Producto {
        return this._producto;
    }

    public set $producto(value: Producto) {
        this._producto = value;
    }

    public get $almacen(): Almacen {
        return this._almacen;
    }

    public set $almacen(value: Almacen) {
        this._almacen = value;
    }

    public get $cliente(): Cliente {
        return this._cliente;
    }

    public set $cliente(value: Cliente) {
        this._cliente = value;
    }
}